app/views/topology/service.js

```diff
@@ -1346,6 +1349,29 @@ YUI.add('juju-topology-service', function(Y) {
     },
 
     /**
+      Returns a service name which is a maximum of 10 characters wide.
+
+      @method truncateServiceName
+      @param {Object} service The service model
+      @return {String} The truncated service display name.
+    */
+    truncateServiceName: function(service) {
+      var name = service.displayName;
```

Comment:
ah no, ok so this is the service. I wonder about not passing the whole model here.